product-service/src/resources/products/product.controller.ts

```diff
@@ -0,0 +1,25 @@
+import {ProductService} from "./product.service";
+import {successResponse, errorResponse, notFoundResponse} from "@libs/apiGateway";
+
+export class ProductController {
```

Comment:
Good approach with 3-layer architecture!
But names of classes (`ProductService`, `ProductControlled`) are little bit confusing, because usually `controller` userd in the name of lowest level of layer (Data Access Level sometimes) and `service` in the middle level of layer (connection between API layer and DAL level).
But it doesn't affect proper work of service, of course - just a little point from my side 🙂 

Attach few links to articles, but I believe you already familiar with that architecture 😃 
https://dev.to/santypk4/bulletproof-node-js-project-architecture-4epf
https://bytearcher.com/articles/node-project-structure/